### PR TITLE
Rename providers to sockets in `lib/sdf-server` and clean up what remains

### PR DESCRIPTION
**Pull request by nickgerace**
## Description

This PR renames providers to sockets in `lib/sdf-server` and cleans up what remains. We have left some usages of "provider" live on where it makes sense: like commented out code in currently active areas.

This commit also restores some `sdf` tests and removes unused `AttributeContext` and `AttributeReadContext` code. Woot woot!

## Additional Changes

- Remove unused `/api/provider` module and route (side note: why was this here if it was unused?)
- Remove dead or commented out code in `WsEvent` file, but keep commented out variants for now
- Restore force change set when creating connections
- Optimize schema variant socket retrieval in `sdf_mock` integration test

## Context

This is a follow-up to https://github.com/systeminit/si/pull/3388, but focuses on `lib/sdf-server` rather than `lib/dal`.

### CODE REVIEW

File: lib/dal/src/attribute/prototype.rs [hunk 1]
```diff
@@ -301,8 +297,6 @@ impl AttributePrototype {
             for attribute_value_target in workspace_snapshot
                 .incoming_sources_for_edge_weight_kind(target_id, edge_weight_discrim)?
             {
-                // There are also provider edges from the schema variant to the provider.
-                // These should be different edge kinds, I think
```
Comment (nickgerace) [inline lib/dal/src/attribute/prototype.rs:305]: @zacharyhamm I believe this was your comment. Worth deleting?

Comment (zacharyhamm) [reply lib/dal/src/attribute/prototype.rs:305]: yes, nuke it

**Comment by nickgerace** [comment]
bors merge

**Comment by nickgerace** [comment]
bors cancel

**Comment by nickgerace** [comment]
bors merge

**Pull request merged**